src/polkadot/polkadotInfo.ts

```diff
@@ -45,6 +46,12 @@ export const polkadot = makeOuterPlugin<{}, PolkadotTools>({
   currencyInfo,
   networkInfo: {},
 
+  checkEnvironment: () => {
+    if (typeof BigInt(1) !== 'bigint') {
+      throw new Error('Environment error: bigint unsupported')
+    }
+  },
```

Comment:
As mentioned in the other core PR, this is not the right design. We want the Polkadot plugin to exist, but simply fail when creating the tools or engine. This allows users to see that their wallets exist, but simply can't load on this particular phone. That way, they don't call support saying their funds disappeared if they switch phones or something.

One thing that will improve the UX is to make the wallet list loading row capable of showing an error message. Instead of a spinner, the dead Polkadot wallet would then show a message like "Not supported on this phone", so it's clear what happened. This would be powered by the new `EdgeAccount.currencyWalletErrors`

However, that's not the goal right away. The goal is just to have the red drop-down show something more meaningful.

Reply:
After many iterations (and foot guns) I was able to get this to fail late with a custom error message